.github/workflows/ci.yml

```
@@ -1,181 +1,148 @@
+      env:
+        DATABASE_URL: sqlite:///./test.db
+        SECRET_KEY: test-secret-key
+        JWT_SECRET_KEY: test-jwt-secret
       run: |
-        cd backend
-        pytest tests/test_ddos_dns_protection.py -v --cov=app.middleware.ddos_protection --cov=app.core.dns_security --cov-report=term-missing
-        echo "DDoS/DNS Protection tests completed"
+        pytest tests/ -v --cov=app --cov-report=xml --cov-report=term
     
     - name: Upload coverage
       uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)